### ISSUE TITLE
SPE1CASE2_ACTNUM testcase oscillations

### ISSUE THREAD

**Issue by atgeirr** [OPM/opm-simulators#1161]
As currently in opm-data the case stops after 1 year. If extending it (with for example 350 more 10-day timesteps) the WGIR for INJ develops some oscillations between approx. 2000-3000 days.

We should investigate this after the release.

**Comment by andlaus** [comment]
which simulator are you using? I just tested this with flow_ebos, and it seems to work fine...

**Comment by GitPaean** [comment]
The ECL result has more than 3000 days, while the DATA only contain about 1 years. 

If you extend the DATA file to be more than 3000 days, you will see the the oscillation in the well curves.

**Comment by atgeirr** [comment]
> which simulator are you using?

All of them exhibit the same behaviour.

**Comment by andlaus** [comment]
ah, okay. I did not realize that the deckfile must be modified. but before we proceed: @totto82 or @joakim-hove, can you check if this issue does also occur with E100?

**Comment by atgeirr** [comment]
> can you check if this issue does also occur with E100?

For this case we have Eclipse results for a longer period, and they do not show oscillations.

**Comment by andlaus** [comment]
yes, but maybe something else changed in the deckfile. e.g.: what's the schedule for the "extension"?

**Comment by andlaus** [comment]
(I just tried with month long steps over 10 years: no problems.)

**Comment by andlaus** [comment]
Well, there's one convergence failure after 1945 days that causes a time step reduction, but according to the terminal output it did not detect any oscillations. I am not sure whether it is worthwhile to investigate this because the residual is known to be non-smooth due to upwinding and due to using piecewise linear material laws anyway, so occasional convergence failures ought to be expected...

**Comment by GitPaean** [comment]
Please refer to the WGIR curve for INJ in the following link. 

https://www.dropbox.com/s/056exfy123uqjn9/spe1case2_actnum.pdf?dl=0

**Comment by andlaus** [comment]
> For this case we have Eclipse results for a longer period, and they do not show oscillations.

yeah, these curves indeed look weird with OPM. did E100 use the same schedule?

**Comment by GitPaean** [comment]
> yeah, these curves indeed look weird with OPM. did E100 use the same schedule?

The TSTEP from the ECL, please ignore the line numbers.

```
  454  0: TSTEP
  455  0: --Advance the simulater once a month for TEN years:
  456  0: 31 28 31 30 31 30 31 31 30 31 30 31
  457  0: 31 28 31 30 31 30 31 31 30 31 30 31
  458  0: 31 28 31 30 31 30 31 31 30 31 30 31
  459  0: 31 28 31 30 31 30 31 31 30 31 30 31
  460  0: 31 28 31 30 31 30 31 31 30 31 30 31
  461  0: 31 28 31 30 31 30 31 31 30 31 30 31
  462  0: 31 28 31 30 31 30 31 31 30 31 30 31
  463  0: 31 28 31 30 31 30 31 31 30 31 30 31
  464  0: 31 28 31 30 31 30 31 31 30 31 30 31
  465  0: 31 28 31 30 31 30 31 31 30 31 30 31 /
```

**Comment by andlaus** [comment]
Ok, I agree that this should be investigated after the release.

**Comment by totto82** [comment]
The problem occur when gas is forced to take a dip due to the missing cells. The two ACTNUM =0 rows: x = 2, y = 1:10, z = 1 and x=1:10, y=2, z=1 is blocking the flow in the top layer. The gas saturation is oscillating after the dip. (I have moved around the blocking rows and observe the same behavior)

**Comment by GitPaean** [comment]
This issue still applies for the current master.

**Comment by blattms** [comment]
@GitPaean Still an issue?
